diff --git a/components/Code.astro b/components/Code.astro
index f0cb26326516686554a6136da4ab84b6dc70184f..309cb6e65f2668b93799fa3a54afb2511cdaaeb0 100644
--- a/components/Code.astro
+++ b/components/Code.astro
@@ -6,6 +6,7 @@ import type {
 	SpecialLanguage,
 	ThemeRegistration,
 	ThemeRegistrationRaw,
+	ShikiTransformer,
 } from 'shiki';
 import { bundledLanguages } from 'shiki/langs';
 import { getCachedHighlighter } from '../dist/core/shiki.js';
@@ -50,6 +51,12 @@ interface Props extends Omit<HTMLAttributes<'pre'>, 'lang'> {
 	 * @default false
 	 */
 	inline?: boolean;
+	/**
+	 * Shiki transformers to customize the generated HTML by manipulating the hast tree.
+	 * Supports all transformers listed here: https://shiki.style/packages/transformers#transformers
+	 * Instructions for custom transformers: https://shiki.style/guide/transformers
+	 */
+	transformers?: ShikiTransformer[];
 }
 
 const {
@@ -59,6 +66,7 @@ const {
 	themes = {},
 	wrap = false,
 	inline = false,
+	transformers = [],
 	...rest
 } = Astro.props;
 
@@ -85,6 +93,7 @@ const highlighter = await getCachedHighlighter({
 	theme,
 	themes,
 	wrap,
+	transformers,
 });
 
 const html = await highlighter.highlight(code, typeof lang === 'string' ? lang : lang.name, {
